### DISCUSSION TITLE
Regression in V0.2.0post3?

### DISCUSSION THREAD

**Discussion by RBEmerson970** [remsky/Kokoro-FastAPI#141] [Q&A]
Installing v0.2.0post3 appears introduce a regression. Through post2, FastKoko works as expected. With post3, clicking on Generate Speech, there's an error (See screen grab).
![2025-02-08 12_58_43-FastKoko_ Kokoro-based TTS - Vivaldi](https://github.com/user-attachments/assets/1fe75615-e5ac-4457-9a1a-a6f604f6d6ad)
Here's the CMD contents:
<c:\Users\pavil>docker run --gpus all -p 8880:8880 ghcr.io/remsky/kokoro-fastapi-gpu:v0.2.0post3

==========
== CUDA ==
==========

CUDA Version 12.4.1

Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.

2025-02-08 17:44:56.471 | INFO     | __main__:download_model:60 - Model files already exist and are valid
   Building kokoro-fastapi @ file:///app
      Built kokoro-fastapi @ file:///app
Uninstalled 1 package in 1ms
Installed 1 package in 0.63ms
INFO:     Started server process [67]
INFO:     Waiting for application startup.
05:45:06 PM | INFO     | main:56 | Loading TTS model and voice packs...
05:45:06 PM | INFO     | model_manager:38 | Initializing Kokoro V1 on cuda
05:45:06 PM | DEBUG    | paths:102 | Searching for model in path: /app/api/src/models
05:45:06 PM | INFO     | kokoro_v1:44 | Loading Kokoro model on cuda
05:45:06 PM | INFO     | kokoro_v1:45 | Config path: /app/api/src/models/v1_0/config.json
05:45:06 PM | INFO     | kokoro_v1:46 | Model path: /app/api/src/models/v1_0/kokoro-v1_0.pth
/app/.venv/lib/python3.10/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
  warnings.warn(
/app/.venv/lib/python3.10/site-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
05:45:08 PM | DEBUG    | paths:154 | Scanning for voices in path: /app/api/src/voices/v1_0
05:45:08 PM | DEBUG    | paths:132 | Searching for voice in path: /app/api/src/voices/v1_0
05:45:08 PM | DEBUG    | model_manager:77 | Using default voice 'af_heart' for warmup
05:45:08 PM | INFO     | kokoro_v1:75 | Creating new pipeline for language code: a
05:45:09 PM | DEBUG    | kokoro_v1:230 | Generating audio for text with lang_code 'a': 'Warmup text for initialization....'
05:45:10 PM | DEBUG    | kokoro_v1:238 | Got audio chunk with shape: torch.Size([57600])
05:45:10 PM | INFO     | model_manager:84 | Warmup completed in 3706ms
05:45:10 PM | INFO     | main:97 |

░░░░░░░░░░░░░░░░░░░░░░░░

    ╔═╗┌─┐┌─┐┌┬┐
    ╠╣ ├─┤└─┐ │
    ╚  ┴ ┴└─┘ ┴
    ╦╔═┌─┐┬┌─┌─┐
    ╠╩╗│ │├┴┐│ │
    ╩ ╩└─┘┴ ┴└─┘

░░░░░░░░░░░░░░░░░░░░░░░░

Model warmed up on cuda: kokoro_v1CUDA: True
67 voice packs loaded

Beta Web Player: http://0.0.0.0:8880/web/
or http://localhost:8880/web/
░░░░░░░░░░░░░░░░░░░░░░░░

INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8880 (Press CTRL+C to quit)
INFO:     172.17.0.1:38026 - "GET /web HTTP/1.1" 307 Temporary Redirect
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38026 - "GET /web/ HTTP/1.1" 200 OK
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38026 - "GET /web/styles/base.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38040 - "GET /web/styles/layout.css HTTP/1.1" 200 OK
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38084 - "GET /web/styles/responsive.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38054 - "GET /web/styles/header.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38064 - "GET /web/styles/forms.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38080 - "GET /web/styles/player.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38026 - "GET /web/styles/badges.css HTTP/1.1" 200 OK
INFO:     172.17.0.1:38040 - "GET /web/styles/controls.css HTTP/1.1" 200 OK
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38084 - "GET /web/src/App.js HTTP/1.1" 200 OK
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38084 - "GET /web/src/services/AudioService.js HTTP/1.1" 200 OK
INFO:     172.17.0.1:38064 - "GET /web/src/components/PlayerControls.js HTTP/1.1" 200 OK
INFO:     172.17.0.1:38080 - "GET /web/src/state/PlayerState.js HTTP/1.1" 200 OK
INFO:     172.17.0.1:38040 - "GET /web/src/services/VoiceService.js HTTP/1.1" 200 OK
INFO:     172.17.0.1:38054 - "GET /web/src/components/WaveVisualizer.js HTTP/1.1" 200 OK
INFO:     172.17.0.1:38026 - "GET /web/src/components/VoiceSelector.js HTTP/1.1" 200 OK
05:47:00 PM | DEBUG    | paths:310 | Searching for web file in path: /app/web
INFO:     172.17.0.1:38084 - "GET /web/src/components/TextEditor.js HTTP/1.1" 200 OK
05:47:00 PM | INFO     | openai_compatible:61 | Created global TTSService instance
05:47:00 PM | DEBUG    | paths:154 | Scanning for voices in path: /app/api/src/voices/v1_0
INFO:     172.17.0.1:38084 - "GET /v1/audio/voices HTTP/1.1" 200 OK
05:48:41 PM | DEBUG    | paths:154 | Scanning for voices in path: /app/api/src/voices/v1_0
INFO:     172.17.0.1:35706 - "POST /v1/audio/speech HTTP/1.1" 200 OK
05:48:41 PM | DEBUG    | paths:154 | Scanning for voices in path: /app/api/src/voices/v1_0
05:48:41 PM | INFO     | openai_compatible:133 | Starting audio generation with lang_code: None
05:48:41 PM | DEBUG    | paths:132 | Searching for voice in path: /app/api/src/voices/v1_0
05:48:41 PM | DEBUG    | tts_service:210 | Using single voice path: /app/api/src/voices/v1_0/af_alloy.pt
05:48:41 PM | DEBUG    | tts_service:234 | Using voice path: /app/api/src/voices/v1_0/af_alloy.pt
05:48:41 PM | INFO     | tts_service:238 | Using lang_code 'a' for voice 'af_alloy' in audio stream
05:48:41 PM | INFO     | text_processor:97 | Starting smart split for 133 chars
Error processing file '/home/runner/work/espeakng-loader/espeakng-loader/espeak-ng/_dynamic/share/espeak-ng-data/phontab': No such file or directory.
c:\Users\pavil>>

**Comment by RBEmerson970** [comment]
See #138 for a fix to this problem.
